internal/service/cloud/server.go

```diff
@@ -334,7 +347,11 @@ func (s *Service) createServer(ctx context.Context, secret *corev1.Secret, ms *s
 		return fmt.Errorf("image lookup: %w", err)
 	}
 
-	renderedData := s.renderUserData(ms, string(bootstrapData))
+	renderedData, err := s.enrichUserData(ms, bootstrapFormat, string(bootstrapData))
+	if err != nil {
+		return fmt.Errorf("enriching user data: %w", err)
```

Comment:
We can improve user visibility here.
by setting a condition and throwing a warning event.

Reply:
What exactly do you mean by setting a condition?

Reply:
something like:
```
conditions.MarkFalse(
	ms.IonosMachine,
	infrav1.MachineProvisionedCondition,
	infrav1.RenderBootstrapDataReason,
	clusterv1.ConditionSeverityInfo, "",
)
```